superset/commands/dataset/update.py

```diff
@@ -86,15 +86,21 @@ def validate(self) -> None:
         except SupersetSecurityException as ex:
             raise DatasetForbiddenError() from ex
 
-        database_id = self._properties.get("database", None)
-        table_name = self._properties.get("table_name", None)
+        database_id = self._properties.get("database")
+
+        table = Table(
+            self._properties.get("table_name"),
+            self._properties.get("schema"),
+            self._model.catalog,
+        )
+
         # Validate uniqueness
         if not DatasetDAO.validate_update_uniqueness(
             self._model.database_id,
-            Table(table_name, self._model.schema, self._model.catalog),
```

Comment:
Here's the bug. The schema should be the schema defined in the properties and not the schema associated with the model.